server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -211,6 +212,31 @@ public static <E> List<E> appendToCopy(Collection<E> collection, E element) {
         return Collections.unmodifiableList(Arrays.asList(array));
     }
 
+    /**
+     * Creates a copy of the given set and adds extra element.
+     *
+     * @param set     set to copy
+     * @param element element to add
+     */
+    public static <E> Set<E> addToCopy(Set<E> set, E element) {
```

Comment:
A common pattern I've seen in the codebase is to use the Streams API for appending an element to a collection:
```java
Stream.concat(set.stream(), Stream.of(element)).collect(Collectors.toUnmodifiableSet())
```

Might be not the fastest way, but I'm not sure how much we care about performance here.

Reply:
Yeah. I only changed a couple of places where this is executed during a start up on a fairly small sets. May be it would be preferable to have them as ImmutableSetN rather then a HashSet wrapped into UnmodifyableSet.